src/lib/device-types/index.ts

```diff
@@ -291,61 +248,59 @@ function syncDataModel(
 		);
 		return;
 	}
-	return sbvrUtils.db.transaction(tx => {
-		return Promise.each(Object.values(types), deviceTypeInfo => {
-			const deviceType = deviceTypeInfo.latest.deviceType;
-			return updateDTModel(deviceType, propertyMap, tx);
-		});
+	return sbvrUtils.db.transaction(async tx => {
+		for (const key in types) {
+			const { deviceType } = types[key].latest;
+			await updateDTModel(deviceType, propertyMap, tx);
+		}
 	});
 }
 
-function fetchDeviceTypesAndReschedule(): Promise<Dictionary<DeviceTypeInfo>> {
-	const promise = fetchDeviceTypes()
-		.tap(() => {
-			// when the promise gets resolved, cache it
-			deviceTypesCache = promise;
-		})
-		.finally(() => {
+let deviceTypesCache: Promise<Dictionary<DeviceTypeInfo>> | undefined;
+
+async function fetchDeviceTypesAndReschedule(): Promise<
+	Dictionary<DeviceTypeInfo>
+> {
+	const promise = fetchDeviceTypes().finally(async () => {
+		try {
 			// schedule a re-run to update the local cache
-			Promise.delay(DEVICE_TYPES_CACHE_EXPIRATION)
-				.then(fetchDeviceTypesAndReschedule)
-				.catch(err => {
-					captureException(err, 'Failed to re-fetch device types');
-				});
+			await Bluebird.delay(DEVICE_TYPES_CACHE_EXPIRATION);
+			await fetchDeviceTypesAndReschedule();
+		} catch (err) {
+			captureException(err, 'Failed to re-fetch device types');
+		}
 
-			// silence the promise created but not returned warning
-			return null;
-		});
+		// silence the promise created but not returned warning
+		return null;
+	});
 
 	// if the cache is still empty, use this promise so that
 	// we do not start a second set of requests to s3
 	// in case another api request comes before the first completes
 	if (!deviceTypesCache) {
 		deviceTypesCache = promise;
-	} else {
-		updateDeviceTypesCache(promise);
```

Comment:
I removed the `updateDeviceTypesCache` function altogether because all it does is mutate the contents of the previous cache with the updated info but as we don't keep the previously resolved cache contents around anywhere there's no need to mutate it and so we can simplify this and just rely on the
```ts
// when the promise gets resolved, cache it
deviceTypesCache = promise;
```
below to update the cache